### PR TITLE
Convert mob_db.txt to mob_db.yml format

### PR DESCRIPTION
**Pull request by lukasrmattos**
I know there exists a old PR with this, but the owner was busy and not finished the PR and didn't update with new yaml logic. Of course if the way I made is not as you guys desire, feel free to reject this pr.

I would like to hear suggestions about the mob_db layout I choose (check mob_db.yml documentation) and where to improve.

I would appreciate if someone make all tests about mob right load.

- [X] Mob_db.txt Re conversion to yml
- [X] Mob_db.txt Pre-Re conversion to yml
- [ ] csv2yml

By the way, I'm excluding SQL mob load until I think best approach than use old method for this.

### CODE REVIEW

**Comment by lukasrmattos** [comment]
I don't know why AppVeyor is saying that build failed (the console is fine and mine is compiling too). How to fix it?

**Comment by Lemongrass3110** [comment]
https://ci.appveyor.com/project/rAthenaAPI/rathena/builds/33283338/job/6yvlgw1o43hf93xd#L863

```
[Status]: Done reading '1975' entries in 'db/re/mob_db.yml'
[Error]: Failed to read MOB_DB database file from 'db/import/mob_db.yml'.
[Error]: bad file (Line -1: Column -1)
```

You forgot to add the copying from import-tmpl -> import in the visual studio project.

**Comment by lukasrmattos** [comment]
> https://ci.appveyor.com/project/rAthenaAPI/rathena/builds/33283338/job/6yvlgw1o43hf93xd#L863
> 
> ```
> [Status]: Done reading '1975' entries in 'db/re/mob_db.yml'
> [Error]: Failed to read MOB_DB database file from 'db/import/mob_db.yml'.
> [Error]: bad file (Line -1: Column -1)
> ```
> 
> You forgot to add the copying from import-tmpl -> import in the visual studio project.

Fixed, thanks.

**Comment by admkakaroto** [comment]
It is possible to create an 'tool' for converte mob_db.txt into yml?

File: src/map/mob.cpp [hunk 1]
```diff
@@ -4187,188 +4290,663 @@ static bool mob_parse_dbrow(char** str)
...
+
+		if (this->nodeExists(modeNode, "IgnoreMiscSkills"))
+		{
+			if (this->asBool(modeNode, "IgnoreMiscSkills", mode))
+				status->mode = mode ? static_cast<enum e_mode>(status->mode | MD_IGNOREMISC) : static_cast<enum e_mode>(status->mode & ~MD_IGNOREMISC);
+		}
+		else
+			status->mode = static_cast<enum e_mode>(status->mode & ~MD_IGNOREMISC);
+
+		if (this->nodeExists(modeNode, "KockbackImmune"))
```
Comment (Latiosu) [inline src/map/mob.cpp:4684]: @lukasrmattos Is it possible to rename this to `KnockbackImmune`?

**Comment by lukasrmattos** [comment]
> It is possible to create an 'tool' for converte mob_db.txt into yml?

I was wrapped up with some services, I'll see if I do this weekend. I still don't fully understand the emulator csv2yml, I have to stop to read it, I did mine in C #

File: src/map/mob.cpp (hunk 1 above)
Comment (lukasrmattos) [reply src/map/mob.cpp]: Done

**Comment by Lemongrass3110** [comment]
Closed in favor of #5435 - thank you for your contribution!

**Pull request closed without merging**